### PR TITLE
drop support for old Symfony versions

### LINKED ISSUE #51: Drop support for Symfony 2.3 to 2.6
Symfony 3.3 and 4.0 are right around the corner and symfony 2.3 is no longer supported we should drop it to.

As suggested by @garak we should also add tests `composer update --prefer-lowest` to check the lowest possible version

### PR DESCRIPTION
**Pull request by garak**
This should fix #51 
This PR is also sorting dependencies.

### CODE REVIEW

**Comment by cmodijk** [comment]
@garak Thanks for taking the time to make this pull request. Can you take a look at the travis build? I seems to be failing.

**Comment by garak** [comment]
I made some changes and squashed. Now, as you can see in travis, there's a small problem with php 5.6, and the error is a bit weird: `The service definition "service_container" does not exist`.
Any help will be appreciated

**Comment by ruudk** [comment]
It's related to this issue: https://github.com/symfony/symfony/pull/18893

We should change our composer.json requirement:
```diff
-"symfony/dependency-injection": "~2.7 || ~3.3 || ~4.0",
+"symfony/dependency-injection": "~2.7.16 || ~3.3 || ~4.0",
```

It will proceed further, but still fails later on with a different issue.

**Comment by ruudk** [comment]
The second issue is fixed when require a minimum version of 3.0.1 for message-bus
```diff
-"simple-bus/message-bus": "~3.0",
+"simple-bus/message-bus": "~3.0.1",
```

**Comment by garak** [comment]
It's all green now! :green_heart:

File: composer.json [hunk 1]
```diff
@@ -18,26 +18,26 @@
...
     "require-dev": {
+        "doctrine/doctrine-bundle": "~1.7",
+        "doctrine/orm": "~2.3",
         "phpunit/phpunit": "^5.7 || ^6.0",
         "simple-bus/doctrine-orm-bridge": "~5.0",
-        "doctrine/orm": "~2.3",
-        "doctrine/doctrine-bundle": "~1.0",
-        "symfony/monolog-bridge": "~2.3 || ~3.0 || ~4.0",
-        "symfony/monolog-bundle": "~2.3 || ~3.0 "
+        "symfony/monolog-bridge": "~2.8 || ~3.2 || ~4.0",
```
Comment (ruudk) [inline composer.json:39]: Shouldn't this be 3.3?

Comment (garak) [reply composer.json:39]: It looks like monolog-bridge is not following the same versions of Symfony.

Comment (ruudk) [reply composer.json:39]: :+1:

**Review by ruudk — APPROVED** [review]
Looks good, thanks for this! 🙌  @cmodijk Merge :) ?

**Comment by ruudk** [comment]
@garak Now that https://github.com/SimpleBus/SymfonyBridge/pull/71 is merged, can you rebase this PR?

**Comment by ruudk** [comment]
We'll close this PR because we are moving SimpleBus to a monorepo. We'll take care of it there.

**Pull request closed without merging**